detection_rules/rule_validators.py

```diff
@@ -657,13 +661,16 @@ def validate_query_text_with_schema(  # noqa: PLR0913
             # If the error is an unknown field and the field was referenced as optional (prefixed with '?'),
             # treat this target as non-fatal to honor EQL optional semantics.
 
+            # To support EQL sequence and sub query validation we need to return this field to overwrite
+            # what would have been parsed via auto_add_field as the error message and query may be our of sync
+            # depending on how the method is called.
             field = extract_error_field(query_text, exc)
             if (
                 field
                 and ("Unknown field" in message or "Field not recognized" in message)
                 and f"?{field}" in self.query
```

Comment:
🤔 small nit/question: Does this have to be:
 
```suggestion
                and f"?{field}" in self.query_text
```